src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,8 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating())
+            storage.save(addressBook);
```

Comment:
[Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation:exclamation: One or more of these problems noticed: 
1. extra/missing spaces.
2. brace style problem.
3. missing braces.
4. name doesn't follow naming convention